File: tests/test-terraria/src/main/java/pize/tests/terraria/indev/TileTextures.java
package pize.tests.terraria.indev;

import pize.graphics.texture.Texture;

public class TileTextures{

    private final Texture texture;

    private TileTextures(Texture texture){
        this.texture = texture;
    }



    public static TileTextures generate(Texture texture, TextureType type){
        TileTextures tileTextures = new TileTextures(texture);

        if(type == TextureType.TILE_DEFAULT){

        }

        return tileTextures;
    }

}

File: tests/test-terraria/src/main/kotlin/pize/tests/terraria/indev/TileTextures.kt
package pize.tests.terraria.indev

import pize.graphics.texture.Texture

class TileTextures private constructor(private val texture: Texture?) {
    companion object {
        fun generate(texture: Texture?, type: TextureType): TileTextures {
            val tileTextures = TileTextures(texture)
            if (type == TextureType.TILE_DEFAULT) {
            }
            return tileTextures
        }
    }
}
